tests/helpers/in-memory-ipfs.ts

```diff
@@ -18,6 +18,7 @@ export async function createInMemoryIPFS(): Promise<IPFS> {
   const memoryDs = new MemoryDatastore()
   const memoryBs = new MemoryBlockstore()
 
+  // @ts-ignore
```

Comment:
Why is that needed? Do we have to fix something in the config?

Reply:
I'm getting a type error for a weird multiformats inconsistency between the dag-pb and ipfs-core libraries 🤔 
```
✖ ERROR: tests/helpers/in-memory-ipfs.ts:21:3 - error TS2322: Type 'IPFS' is not assignable to type 'IPFS<{}>'.
  The types of 'object.put' are incompatible between these types.
    Type '(obj: import("/Users/icid/Work/webnative/node_modules/ipfs-core/node_modules/@ipld/dag-pb/types/src/interface").PBNode, options?: import("/Users/icid/Work/webnative/node_modules/ipfs-core-types/types/src/object/index").PutOptions | undefined) => Promise<...>' is not assignable to type '(obj: import("/Users/icid/Work/webnative/node_modules/@ipld/dag-pb/types/src/interface").PBNode, options?: import("/Users/icid/Work/webnative/node_modules/ipfs-core-types/types/src/object/index").PutOptions | undefined) => Promise<...>'.
      Types of parameters 'obj' and 'obj' are incompatible.
        Type 'import("/Users/icid/Work/webnative/node_modules/@ipld/dag-pb/types/src/interface").PBNode' is not assignable to type 'import("/Users/icid/Work/webnative/node_modules/ipfs-core/node_modules/@ipld/dag-pb/types/src/interface").PBNode'.
          Types of property 'Links' are incompatible.
            Type 'import("/Users/icid/Work/webnative/node_modules/@ipld/dag-pb/types/src/interface").PBLink[]' is not assignable to type 'import("/Users/icid/Work/webnative/node_modules/ipfs-core/node_modules/@ipld/dag-pb/types/src/interface").PBLink[]'.
              Type 'import("/Users/icid/Work/webnative/node_modules/@ipld/dag-pb/types/src/interface").PBLink' is not assignable to type 'import("/Users/icid/Work/webnative/node_modules/ipfs-core/node_modules/@ipld/dag-pb/types/src/interface").PBLink'.
                Types of property 'Hash' are incompatible.
                  Type 'import("/Users/icid/Work/webnative/node_modules/multiformats/types/src/cid").CID' is not assignable to type 'import("/Users/icid/Work/webnative/node_modules/peer-id/node_modules/multiformats/types/cid").CID'.
                    Types have separate declarations of a private property 'asCID'.
```

Reply:
I added a commit that fixes this.
Behind the scenes: I took peek into the `yarn.lock` file. Among all installed libraries we see multiformats twice:
```
multiformats@^9.0.0, multiformats@^9.0.2, multiformats@^9.0.4, multiformats@^9.1.0, multiformats@^9.1.2, multiformats@^9.4.1, multiformats@^9.4.2, multiformats@^9.4.5, multiformats@^9.4.7:
  version "9.4.8"
  resolved "https://registry.yarnpkg.com/multiformats/-/multiformats-9.4.8.tgz#46f74ec116f7871f2a334cc6d4ad3d810dbac341"
  integrity sha512-EOJL02/kv+FD5hoItMhKgkYUUruJYMYFq4NQ6YkCh3jVQ5CuHo+OKdHeR50hAxEQmXQ9yvrM9BxLIk42xtfwnQ==

multiformats@^9.4.13, multiformats@^9.5.4:
  version "9.5.9"
  resolved "https://registry.yarnpkg.com/multiformats/-/multiformats-9.5.9.tgz#99ac8f4b997ec3dd2a2f563bdf8fdebdca437b99"
  integrity sha512-G3lKuMEV3H/414CnhiQr/BYPPacno3modE7ory2VFngxpMNAhg3GAYUeKGhdbllHPxX2Fw7mRjYB9mHO7yUwzQ==
```
It shows it's resolved to version `9.4.8` once and then `9.5.9` another time.
But all the version constraints are above: Everything that caused version `9.4.8` to install, would've worked fine with version `9.5.9` as well in terms of version constraints.
The reason I suspect this happens is because previously we used to not have the `^9.5.4` constraints before (at that time, that version probably didn't exist yet), because they're coming from the new `@ipld/*` libraries. The lockfile tries to prevent installing newer libraries, so instead it just forces two different versions.

The way I fixed this was I just set `resolutions: { multiformats: "9.6.2" }` in the `package.json` file, ran `yarn` and the removed the resolutions thing from the `package.json` again.
That forces yarn to use a single version for all dependencies. The dependencies are fine with that, because the new version is in their version bounds, I just have to force yarn to *not* lock the dependencies once and now they're locked at this compatible version again.

Running `yarn` (and not having `resolutions` in the `package.json`) totally works now & we don't need `ts-ignore` :)

Reply:
Huh ok, thanks! I would've assumed `yarn` would override the lock when you update a dependency. Or maybe `pnpm` behaves differently in this regard, and I'm just confusing the two 🤔